### PR TITLE
Dimension names redux

### PR DESCRIPTION
**Pull request by timspainNERSC**
# Dimension names redux
## Fixes \#550 (again)

---
# Change Description

Accept both the new (`xdim`, `ydim`, `zdim`) and old (`x`, `y`, `z`) dimension names to increase compatibility.

---
# Test Description

Added a test that reads in a file with the old dimension names are ensures the dimensions are still set correctly.

---
# Documentation Impact

No impact. This is a convenience detail and no users should be writing new restart files using the old dimension names.

---
# Other Details

The generated `old_names.nc` file will suffer the same deficiencies as the restart files used to but will never be updated.

### CODE REVIEW

File: core/test/old_names.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    n_coords_comp = datagrp.createDimension("ncoords", n_coords)
+    
+    field_dims = ("y", "x")
+    coord_dims = ("yvertex", "xvertex", "ncoords")
+    zfield_dims = ("z", "y", "x")
+
+    # Array coordinates
+    space = 25000. # 25 km in metres
+    node_x = np.zeros((nfirst + 1, nsecond + 1))
+    node_y = np.zeros((nfirst + 1, nsecond + 1))
```
Comment (jwallwork23) [inline core/test/old_names.py:57]: These don't seem to be used anywhere else.

File: core/test/old_names.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import netCDF4
+import numpy as np
+import numpy.ma as ma
+import time
+import math
```
Comment (jwallwork23) [inline core/test/old_names.py:5]: `ma` and `math` appear to be unused.

**Review by jwallwork23 — APPROVED** [review]
I think I followed the login in `old_names.py` correctly. Looks good to me, there are just a couple of unused modules and arrays that are set to zero and never used.

File: core/test/old_names.py (hunk 1 above)
Comment (timspainNERSC) [reply core/test/old_names.py]: Indeed not, removed.

File: core/test/old_names.py (hunk 2 above)
Comment (timspainNERSC) [reply core/test/old_names.py]: Indeed not, removed.

**Pull request merged**